pkg/controller/cluster/main-controller.go

```diff
@@ -460,6 +500,34 @@ func (c *Controller) Start(threadiness int, stopCh <-chan struct{}) error {
 		select {}
 	}
 
+	// runSTS starts the STS API even if the pod is not the leader
+	runSTS := func(ctx context.Context) {
+		// stsServerWillStart is a channel for the STS Server API
+		stsServerWillStart := make(chan interface{})
+
+		go func() {
+			klog.Infof("Starting STS API server")
+			close(stsServerWillStart)
```

Comment:
`defer close(stsServerWillStart)` so the channel closes when the execution is complete

Reply:
not possible, `c.sts.ListenAndServeTLS` will block the process. If we do `defer close(serverWillStart)` will never happen unless the webserver fails.

We will handle the servers stop in another PR